app/main.py

```diff
@@ -1,3 +1,60 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km):
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other) -> Distance:
+        return Distance(
+            self.km + other.real
+            if isinstance(other, (int, float))
+            else self.km + other.km
+        )
+
+    def __iadd__(self, other) -> Distance:
+        if isinstance(other, (int, float)):
+            self.km += other.real
+        else:
+            self.km += other.km
+        return self
+
+    def __mul__(self, other) -> Distance:
+        return Distance(
```

Comment:
`other` can not be of type `Distance` in this method, because if you multiply km * km you get square km